src/awkward/_broadcasting.py

```diff
@@ -160,7 +160,7 @@ def all_same_offsets(backend: ak._backends.Backend, inputs: list) -> bool:
                 my_offsets = index_nplike.empty(0, dtype=np.int64)
             else:
                 my_offsets = index_nplike.arange(
-                    0, x.content.length, x.size, dtype=np.int64
+                    0, x.content.length + 1, x.size, dtype=np.int64
```

Comment:
We want a range that is _inclusive_ of the endpoints, which is not the case by default.

Reply:
We do? Oh, yes—we do! Offsets always have to be one longer than the array itself; that's right. Good catch!